pkg/api/norman/customization/gke/handler.go

```diff
@@ -0,0 +1,194 @@
+package gke
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/rancher/norman/api/access"
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/pkg/auth/util"
+	client "github.com/rancher/rancher/pkg/client/generated/management/v3"
+	v1 "github.com/rancher/rancher/pkg/generated/norman/core/v1"
+	"github.com/rancher/rancher/pkg/namespace"
+	"github.com/rancher/rancher/pkg/ref"
+	schema "github.com/rancher/rancher/pkg/schemas/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/types/config"
+	"github.com/sirupsen/logrus"
+)
+
+type Capabilities struct {
+	Credentials string `json:"credentials,omitempty"`
+	ProjectID   string `json:"projectId,omitempty"`
+	Zone        string `json:"zone,omitempty"`
+	Region      string `json:"region,omitempty"`
+}
+
+// GKE handler lists available resources in Google API
+type handler struct {
+	Action        string
+	schemas       *types.Schemas
+	secretsLister v1.SecretLister
+}
+
+func NewGKEHandler(scaledContext *config.ScaledContext) http.Handler {
+	return &handler{
+		schemas:       scaledContext.Schemas,
+		secretsLister: scaledContext.Core.Secrets(namespace.GlobalNamespace).Controller().Lister(),
+	}
+}
+
+func (h *handler) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
+
+	writer.Header().Set("Content-Type", "application/json")
+
+	capa := &Capabilities{}
+
+	if credID := req.URL.Query().Get("cloudCredentialId"); credID != "" {
```

Comment:
If gkeV2 we are using a GET call ? Why not a POST ?

Reply:
It is getting `cloudCredentialId` parameter from HTTP request URL

Reply:
We still use POST method

Reply:
I guess the problem is figuring out whether to use the new cloud credential method versus the raw credential JSON? It is a little weird to use POST with URL query parameters. What about using POST but adding `CloudCredentialID` to the `Capabilities` struct and checking that one of either `CloudCredentialID` or `Credentials` is populated?

Reply:
This was implementation was done based on `oci` and `vsphere` module. So probably this would be done on UI site.